core/src/main/java/com/scalar/db/common/AbstractTwoPhaseCommitTransactionManager.java

```diff
@@ -68,9 +74,17 @@ public Optional<String> getTable() {
     return tableName;
   }
 
-  protected TwoPhaseCommitTransaction activate(TwoPhaseCommitTransaction transaction)
+  protected TwoPhaseCommitTransaction decorate(TwoPhaseCommitTransaction transaction)
       throws TransactionException {
-    return new StateManagedTransaction(transaction);
+    TwoPhaseCommitTransaction decorated = transaction;
+    for (TwoPhaseCommitTransactionDecorator transactionDecorator : transactionDecorators) {
+      decorated = transactionDecorator.decorate(decorated);
+    }
+    return decorated;
+  }
+
+  public void addTransactionDecorator(TwoPhaseCommitTransactionDecorator transactionDecorator) {
```

Comment:
We can add a transaction decorator with this method. This method is a `public` scope so that we can also add a transaction decorator externally.